### PR TITLE
feat(step-generation, protocol-designer): add JSON schema v6 support

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview

This PR changes all of step generation and protocol designer to support JSON schema v6 protocols. It also adds some infrastructure to support heater shaker commands in step generation and protocol designer. 

Co-authored by @jerader and @sakibh 

Note: I turned off migration e2e tests for now, ~and there is an e2e test failing because we need to update the PD export modal but that'll be fixed soon~ (fixed now!), so please still review! No need to closely review changes to test files (though if you want to please do!), just please pay carful attention to changes to the command creators and to the `fileCreator` file. 

# Changelog

- Add JSON schema v6 support

# Review requests
- Import a <v6 protocol, export it, and make sure it conforms to the JSON v6 spec 

An easy way to verify this is by downloading the generated PD protocol, moving it into `shared-data/protocol/fixtures/6`, and then running `shared-data/js/__tests__/protocolSchemaV6.test.ts`. That test will make sure all protocols in `shared-data/protocol/fixtures/6` conform to the schema v6 spec

# Risk assessment
High, this changes very foundational logic

### CODE REVIEW

File: protocol-designer/src/file-data/__fixtures__/createFile/v6Fixture.ts [hunk 1]
```diff
@@ -34,173 +34,217 @@ export const robotStateTimeline: Timeline = {
     {
```
Comment (b-cooper) [inline protocol-designer/src/file-data/__fixtures__/createFile/v6Fixture.ts:34]: Does it make sense to keep the v5 fixture around so we can still test that old files are properly migrated to v6?

File: shared-data/protocol/types/schemaV6/command/gantry.ts [hunk 2]
```diff
@@ -82,16 +82,16 @@ interface MoveToSlotParams {
...
 
-interface MoveToWellParams {
+export interface MoveToWellParams {
   pipetteId: string
   labwareId: string
   wellName: string
-  wellLocation: {
-    origin: 'top' | 'bottom'
+  wellLocation?: {
+    origin?: 'top' | 'bottom'
```
Comment (b-cooper) [inline shared-data/protocol/types/schemaV6/command/gantry.ts:90]: should `origin` and `offset` be optional? if you supply `wellLocation`, I would assume we'd expect at least `origin` to be specified if `offset` defaults to 0,0,0

File: step-generation/src/__tests__/deactivateTemperature.test.ts [hunk 3]
```diff
@@ -58,7 +58,7 @@ describe('deactivateTemperature', () => {
     //   expected: {
```
Comment (b-cooper) [inline step-generation/src/__tests__/deactivateTemperature.test.ts]: is this TODO still relevant?

File: protocol-designer/src/file-data/__fixtures__/createFile/v6Fixture.ts (hunk 1 above)
Comment (shlokamin) [reply protocol-designer/src/file-data/__fixtures__/createFile/v6Fixture.ts:34]: this fixture only seemed to be used to test the `createFile` selector (which expects a timeline that holds commands that conform to schema v6 commands). we still have v5 fixtures elsewhere!

File: shared-data/protocol/types/schemaV6/command/gantry.ts (hunk 2 above)
Comment (shlokamin) [reply shared-data/protocol/types/schemaV6/command/gantry.ts:90]: check out this discussion : https://opentrons.slack.com/archives/C02JZQB567R/p1637174176038200?thread_ts=1637084498.009500&cid=C02JZQB567R

those parameters should be assigned defaults by PE

File: step-generation/src/__tests__/deactivateTemperature.test.ts (hunk 3 above)
Comment (shlokamin) [reply step-generation/src/__tests__/deactivateTemperature.test.ts]: doesn't appear to be! i can delete it

**Review by jerader — APPROVED** [review]
smoked tested PD and everything seemed good 🚀

File: step-generation/src/types.ts [hunk 4]
```diff
@@ -373,6 +392,8 @@ export type CommandCreatorArgs =
   | DeactivateTemperatureArgs
   | ThermocyclerProfileStepArgs
   | ThermocyclerStateStepArgs
+  | HeaterShakerArgs
+// | SetShakeSpeedArgs
```
Comment (b-cooper) [inline step-generation/src/types.ts]: should this be commented out?

Comment (shlokamin) [reply step-generation/src/types.ts]: good catch thank you!

Comment (shlokamin) [reply step-generation/src/types.ts]: it should be deleted altogether

**Review by b-cooper — APPROVED** [review]
Looking good ☄️   All of the changes look sound to me

**Pull request merged**